core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -691,11 +692,16 @@ private void loadMetadataForPath(LockedInodePath inodePath)
       throws InvalidPathException, AccessControlException, IOException, FileDoesNotExistException,
       FileAlreadyCompletedException, InvalidFileSizeException, BlockInfoException {
     UfsStatus status = mStatusCache.fetchStatusIfAbsent(inodePath.getUri(), mMountTable);
+    DescendantType descendantType = mDescendantType;
+    if (descendantType.equals(DescendantType.ONE)
```

Comment:
we need to change descendantType to ONE if the original descendantType was not recursive

Reply:
Can you add an explanation as a comment, since all the syncing logic is difficult to follow sometimes.

Reply:
done